crates/core/src/operations/write.rs

```diff
@@ -152,45 +184,61 @@ pub struct WriteBuilder {
     configuration: HashMap<String, Option<String>>,
 }
 
+/// Write data into a DeltaTable
+pub struct WriteBuilder {
+    /// Basic config for the write operation
+    config: WriteBuilderConfig,
+    /// The data to actually write
+    data: Option<WriteData>,
+}
+
 impl WriteBuilder {
     /// Create a new [`WriteBuilder`]
     pub fn new(log_store: LogStoreRef, snapshot: Option<DeltaTableState>) -> Self {
         Self {
-            snapshot,
-            log_store,
-            input: None,
-            state: None,
-            mode: SaveMode::Append,
-            partition_columns: None,
-            predicate: None,
-            target_file_size: None,
-            write_batch_size: None,
-            batches: None,
-            safe_cast: false,
-            schema_mode: None,
-            writer_properties: None,
-            commit_properties: CommitProperties::default(),
-            name: None,
-            description: None,
-            configuration: Default::default(),
+            config: WriteBuilderConfig {
+                snapshot,
+                log_store,
+                state: None,
+                mode: SaveMode::Append,
+                partition_columns: None,
+                predicate: None,
+                target_file_size: None,
+                write_batch_size: None,
+                safe_cast: false,
+                schema_mode: None,
+                writer_properties: None,
+                nr_concurrent_streams: 1,
```

Comment:
```suggestion
                concurrent_streams: 1,
```

We already have the [num_cpus](https://docs.rs/num_cpus/latest/num_cpus/) package as a dependency in core, would it be beneficiail for the default performance case (and validation of this code path) to set concurrent_streams to at least the number of CPUs?

Reply:
I would not. If you write a small file with small batches you'd create lots of very small files otherwise. It would make sense for larger files only